aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -422,13 +431,123 @@ Tensor& clamp_min_out(Tensor& result, const Tensor& self, Scalar min) {
   return result;
 }
 
-Tensor clamp_min(const Tensor& self, Scalar min) {
-  Tensor result = at::empty({0}, self.options());
-  return at::clamp_min_out(result, self, min);
+Tensor clamp_with_tensors(const Tensor& self, const Tensor& min, const Tensor& max) {
+  Tensor result;
+  return at::clamp_with_tensors_out(result, self, min, max);
 }
 
-Tensor& clamp_min_(Tensor& self, Scalar min) {
-  return at::clamp_min_out(self, self, min);
+Tensor clamp_with_tensors_max(const Tensor& self, const Tensor& max) {
+  Tensor result;
+  return at::clamp_with_tensors_max_out(result, self, max);
+}
+
+Tensor clamp_with_tensors_min(const Tensor& self, const Tensor& min) {
+  Tensor result;
+  return at::clamp_with_tensors_min_out(result, self, min);
+}
+
+Tensor& clamp_with_tensors_(Tensor& self, const Tensor& min, const Tensor& max) {
+  return at::clamp_with_tensors_out(self, self, min, max);
+}
+
+Tensor& clamp_with_tensors_out(Tensor& result, const Tensor& self, const Tensor& min, const Tensor& max) {
+  TORCH_CHECK(!self.is_complex() && !min.is_complex() && !max.is_complex(),
+              "clamp does not support complex inputs.");
+  if (min.defined() && max.defined()) {
+    TORCH_CHECK(self.layout() == Layout::Strided,
+                "clamp only supports strided layout, got: ", self.layout());
+    TensorIterator iter = TensorIteratorConfig()
+      .set_check_mem_overlap(true)
+      .add_output(result)
+      .add_input(self)
+      .add_input(min)
+      .add_input(max)
+      .promote_inputs_to_common_dtype(true)
```

Comment:
I would actually disable type promotion and casting to the output for simplicity. It could be enabled separately later.

Reply:
(Otherwise you'll have to add another test for that behavior, but we should focus on landing this.)